### PR TITLE
allow news articles to be redirected

### PR DESCRIPTION
**Pull request by asmega**

### CODE REVIEW

File: spec/controllers/news_controller_spec.rb [hunk 1]
```diff
@@ -30,15 +30,34 @@
...
       it 'raises an ActionController RoutingError' do
-        allow(Core::NewsArticleReader).to receive(:new) { ->(&block) { block.call } }
-
         expect { get :show, id: 'foo', locale: I18n.locale }.to raise_error(ActionController::RoutingError)
       end
     end
+
+    context 'when news article is redirected' do
+      before do
+        allow_any_instance_of(Core::NewsArticleReader).to receive(:call).and_yield(OpenStruct.new(status: 301, location: 'https://example.com', redirect?: true))
```
Comment (Tr4pSt3R) [inline spec/controllers/news_controller_spec.rb]: Give this OpenStruct object a name perhaps? 

```
redirect_path = OpenStruct.new(status: 301, location: 'https://example.com', redirect?: true)
allow_any_instance_of(Core::NewsArticleReader).to receive(:call).and_yield(redirect_path)
```

Comment (munckymagik) [inline spec/controllers/news_controller_spec.rb]: Just to note, and I'm not suggesting you necessarily need to action this as part of this PR, but some of these lines are a bit long, longer than the max width of GitHub's PR diff. As GitHub's diffs don't line-wrap we're having to horizontally scroll to review the entire line.

So yes, intermediate variables or even simply breaking chained statements onto multiple lines would be a big help for review.

**Comment by munckymagik** [comment]
:+1:

File: spec/controllers/news_controller_spec.rb (hunk 1 above)
Comment (asmega) [inline spec/controllers/news_controller_spec.rb]: I've now extracted these to variables which has shortened the lines a bit

**Comment by Tr4pSt3R** [comment]
:+1:

**Pull request merged**